### PR TITLE
FAD-6234 Hide view details button when message_id does property does not exist

### PR DESCRIPTION
**Pull request by rajumsys**
See ticket for the detail context

Didn't find a easy way to manually test it. It would be good if I could modify redux store to modify the data but didn't find a way even though feel like there is a way. 

However, this is pretty trivial and easy enough to verify from snapshot. 

*Update*
Testing:
You can test it using redux dev tool. Click 'Dispatcher'. 

put something like:

```
{
type: 'GET_MESSAGE_EVENTS_SUCCESS',
payload: [{
        formattedDate: 'formatted',
        type: 'injection',
        friendly_from: 'mean@friendly',
        rcpt_to: 'tom.haverford@pawnee.state.in.us',
        message_id: '123abc',
        event_id: '456xyz'
      }, {
        formattedDate: 'formatted',
        type: 'policy_rejection',
        friendly_from: 'mean@friendly',
        rcpt_to: 'tom.haverford@pawnee.state.in.us',
        event_id: '456xyz'
      }]
}
```
And then click tiny little almost microscopic `Dispatch` button on the right. UI should be updated.

### CODE REVIEW

**Comment by jasonrhodes** [comment]
FWIW @rajumsys you can modify redux store using dev tools, you just fire the action you want with the data you want.

**Comment by rajumsys** [comment]
> FWIW @rajumsys you can modify redux store using dev tools, you just fire the action you want with the data you want.

Thanks. I figured out the 'Dispatcher' button after your comment.

File: src/pages/reports/messageEvents/MessageEventsPage.js [hunk 1]
```diff
@@ -50,9 +50,11 @@ export class MessageEventsPage extends Component {
       snakeToFriendly(type),
       rcpt_to,
       friendly_from,
-      <div style={{ textAlign: 'right' }}>
-        <Button onClick={() => this.handleDetailClick({ message_id, event_id })} size='small'>View Details</Button>
-      </div>
+      message_id
```
Comment (jasonrhodes) [inline src/pages/reports/messageEvents/MessageEventsPage.js]: What do you think about moving this into a small component?
```jsx
<ViewDetails messageEvent={rowData}>
```
In there you could wrap it with `withRouter`, so the onClick inline, and exit early if there is no message_id ... would be a bit easier to see what's going on, a bit easier to test clearly, and can get rid of that inline CSS.

Comment (rajumsys) [reply src/pages/reports/messageEvents/MessageEventsPage.js]: it's fine. just test boilerplate :). 

will do

File: src/pages/reports/messageEvents/components/ViewDetailsButton.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  };
+
+  render() {
+    const { message_id, event_id } = this.props;
+
+    if (!message_id) {
+      return null;
+    }
+
+    return (<div style={{ textAlign: 'right' }}>
```
[the remark this answers is not in the document]
Comment (rajumsys) [reply src/pages/reports/messageEvents/components/ViewDetailsButton.js]: moved it as it is. new file? too much work :D

**Pull request merged**